sw-emulator/lib/periph/src/ml_dsa87.rs

```diff
@@ -557,15 +562,20 @@ mod tests {
 
         let mut ml_dsa87 = Mldsa87::new(&clock, key_vault);
 
-        let mut seed = [0u8; 32];
+        let mut seed = rand::thread_rng().gen::<[u8; 32]>();
         seed.to_big_endian(); // Change DWORDs to big-endian. TODO is this needed?
         for i in (0..seed.len()).step_by(4) {
             ml_dsa87
                 .write(RvSize::Word, OFFSET_SEED + i as RvAddr, make_word(i, &seed))
                 .unwrap();
         }
 
-        let mut msg = [0u8; 64];
+        let mut msg: [u8; 64] = {
+            let part0 = rand::thread_rng().gen::<[u8; 32]>();
+            let part1 = rand::thread_rng().gen::<[u8; 32]>();
+            let concat: Vec<u8> = part0.iter().chain(part1.iter()).copied().collect();
+            concat.as_slice().try_into().unwrap()
+        };
         msg.to_big_endian(); // Change DWORDs to big-endian. TODO is this necessary
```

Comment:
Again, this is not necessary as this is a test generated message. Message coming from the hw register will need a DWORD endianness reversal when passing to the FIPS204 sign/verify API.